### PR TITLE
Avatar autogeneration fixed

### PR DESCRIPTION
**Pull request by pboguslawski**
This mod fixes problem with initial avatar autogeneration and
avatar autogneration after deleting previous avatar.

Related: https://github.com/go-gitea/gitea/issues/13159
Fixes: 80a6b0f5bce15a641fc75f5f1ef6e42ef54424bc
Author-Change-Id: IB#1105243

### CODE REVIEW

File: models/user_avatar.go [hunk 1]
```diff
@@ -39,10 +39,9 @@ func (u *User) generateRandomAvatar(e Engine) error {
 	if err != nil {
 		return fmt.Errorf("RandomImage: %v", err)
 	}
-	// NOTICE for random avatar, it still uses id as avatar name, but custom avatar use md5
-	// since random image is not a user's photo, there is no security for enumable
+
 	if u.Avatar == "" {
-		u.Avatar = fmt.Sprintf("%d", u.ID)
+		u.Avatar = base.HashEmail(u.AvatarEmail)
```
Comment (zeripath) [inline models/user_avatar.go:44]: could u.AvatarEmail be "" here?

File: models/user_avatar.go [hunk 2]
```diff
@@ -39,10 +39,9 @@ func (u *User) generateRandomAvatar(e Engine) error {
 	if err != nil {
 		return fmt.Errorf("RandomImage: %v", err)
 	}
-	// NOTICE for random avatar, it still uses id as avatar name, but custom avatar use md5
-	// since random image is not a user's photo, there is no security for enumable
+
 	if u.Avatar == "" {
```
Comment (6543) [inline models/user_avatar.go:43]:
```suggestion
	if len(u.Avatar) == 0 && len(u.AvatarEmail) != 0 {
```

Comment (zeripath) [reply models/user_avatar.go:43]: well if the avatar email is 0 we probably still need to set it to something.

**Comment by pboguslawski** [comment]
base.HashEmail(u.AvatarEmail) was taken from CreateUser; u.AvatarEmail seems to be protected against "":

https://github.com/go-gitea/gitea/blob/e2252a6de150e4b61b067db7abc28927bfe7b085/models/user.go#L821
https://github.com/go-gitea/gitea/blob/e2252a6de150e4b61b067db7abc28927bfe7b085/models/user.go#L190

**Review by zeripath — APPROVED** [review]

**Review by lafriks — APPROVED** [review]

**Comment by lafriks** [comment]
please send backport

**Pull request merged**